### PR TITLE
User report page area chart

### PR DESCRIPTION
**Pull request by inuwa**
## Done
Updated the layout of the user report page to match design

## QA

- Check out this feature branch user 
- Run the site using the command `./run serve`
- View the site locally in your web browser at: [http://0.0.0.0:8001/](http://0.0.0.0:8001/)
- Run through the following [QA steps](https://github.com/canonical-webteam/practices/blob/master/workflow/qa-steps.md)
- go to the url http://0.0.0.0:8001/desktop/statistics
- Check that the layout matches the design https://github.com/canonical-websites/www.ubuntu.com/issues/4094

## Issue / Card
Fixes https://github.com/canonical-websites/www.ubuntu.com/issues/4121

### CODE REVIEW

File: templates/desktop/statistics.html [hunk 1]
```diff
@@ -33,109 +33,138 @@ <h1>Ubuntu user statistics</h1>
...
-    </div>
-    <div class="col-4 p-card--flex">
-      <header class="p-card__header">
-        <h3 class="p-heading--four">Firmware</h3>
-      </header>
-      <div class="p-card__center-content">
-        <svg id="firmware"></svg>
+  <div class="row u-equal-height u-vertically-center p-divider">
+    <div class="col-6 p-divider__block">
+      <div class="col-12">
```
Comment (anthonydillon) [inline templates/desktop/statistics.html]: Please remove this div

File: templates/desktop/statistics.html [hunk 2]
```diff
@@ -33,109 +33,138 @@ <h1>Ubuntu user statistics</h1>
...
+        <h2 class="p-heading--four">OS architecture</h2>
+        <p>OS architecture is responsible for running programs and providing secure access to the machine's hardware.</p>
+      </div>
+      <div class="col-12 u-align--center">
```
Comment (anthonydillon) [inline templates/desktop/statistics.html]: Remove `col-12` please

File: templates/desktop/statistics.html [hunk 3]
```diff
@@ -33,109 +33,138 @@ <h1>Ubuntu user statistics</h1>
...
-      <p class="u-no-margin--bottom">Firmware initialises hardware components and the operating system at start up.</p>
     </div>
-    <div class="col-4 p-card--flex">
-      <header class="p-card__header">
-        <h3 class="p-heading--four">Display server</h3>
-      </header>
-      <div id="display-server" class="p-card__center-content"></div>
-      <p class="u-no-margin--bottom">The display server controls the input and output of its clients to and from the rest of the operating system and the hardware.</p>
+    <div class="col-6 p-divider__block">
+      <div class="col-12">
```
Comment (anthonydillon) [inline templates/desktop/statistics.html]: Remove this div please

File: templates/desktop/statistics.html [hunk 4]
```diff
@@ -33,109 +33,138 @@ <h1>Ubuntu user statistics</h1>
...
+        <h2 class="p-heading--four">Display server</h2>
+        <p>The display server controls the input and output of its clients to and from the rest of the operating system and the hardware.</p>
+      </div>
+      <div class="col-12 u-align--center">
```
Comment (anthonydillon) [inline templates/desktop/statistics.html]: Remove `col-12`

File: templates/desktop/statistics.html [hunk 5]
```diff
@@ -33,109 +33,138 @@ <h1>Ubuntu user statistics</h1>
...
   </div>
-  <div class="row u-equal-height">
-    <div class="col-6 prefix-2">
-      <div class="p-card--flex">
-        <div id="one-screen" class="p-card__center-content"></div>
-        <p class="u-no-margin--bottom">This was detected on install so many users may have connected screens after the installing Ubuntu.</p>
+  <div class="row u-equal-height u-vertically-center p-divider">
+    <div class="col-6 p-divider__block">
+      <div class="row">
+        <div class="col-12 u-align--center">
```
Comment (anthonydillon) [inline templates/desktop/statistics.html]: Please remove this nested row its not needed.

File: templates/desktop/statistics.html [hunk 6]
```diff
@@ -33,109 +33,138 @@ <h1>Ubuntu user statistics</h1>
...
+          <img src="https://via.placeholder.com/350x150" />
+        </div>
+      </div>
+      <div class="row">
```
Comment (anthonydillon) [inline templates/desktop/statistics.html]: You shouldn't nest rows

File: templates/desktop/statistics.html [hunk 7]
```diff
@@ -33,109 +33,138 @@ <h1>Ubuntu user statistics</h1>
...
+          <p>*On install. This was detected on install so many users may have connected screens after the installing Ubuntu.</p>
+        </div>
       </div>
     </div>
-    <div class="col-6 suffix-2">
-      <div class="p-card--flex">
-        <div id="one-gpu" class="p-card__center-content"></div>
-        <p class="u-no-margin--bottom">Graphical Processing Unit (GPU) is used to generate videos and images for displaying on screen.</p>
+    <div class="col-6 p-divider__block">
+      <div class="row">
```
Comment (anthonydillon) [inline templates/desktop/statistics.html]: There is a parent row at line https://github.com/canonical-websites/www.ubuntu.com/pull/4135/files#diff-e60ec2befd8be52ae99dcdc29c3bb7a2R145. You shouldn't need to nest them here.

File: templates/desktop/statistics.html [hunk 8]
```diff
@@ -146,100 +175,83 @@ <h3>Popular screen sizes</h2>
...
-  <div class="row">
-    <h4>Number of CPUs</h4>
-  </div>
-  <div class="row p-mobile-flex-col u-equal-height u-vertically-center">
-    <div class="col-6 u-align--center">
-      <svg id="number-of-cpus" height="320"></svg>
+  <div class="row p-mobile-flex-col u-equal-height p-divider">
+    <div class="col-6 p-divider__block">
+      <h4>Number of CPUs</h4>
+      <div class="row">
```
Comment (anthonydillon) [inline templates/desktop/statistics.html]: Please don't next rows

File: templates/desktop/statistics.html [hunk 9]
```diff
@@ -146,100 +175,83 @@ <h3>Popular screen sizes</h2>
...
-    </div>
-    <div class="col-6">
-      <p>Physical disks denote how much data can be stored on a computer. For this report the total disk size is inferred from the number and size of partitions of a disk.</p>
+      <svg id="physical-disk" height="200"></svg>
     </div>
   </div>
-</section>
-<section class="p-strip is-bordered u-no-padding--top">
-  <div class="row">
+  <div class="row is-bordered">
```
Comment (anthonydillon) [inline templates/desktop/statistics.html]: Does `is-bordered` work on a row?

File: templates/desktop/statistics.html [hunk 10]
```diff
@@ -146,100 +175,83 @@ <h3>Popular screen sizes</h2>
...
+  </div>
+  <div class="row">
+    <div class="col-12">
+      <svg id="partition-type" height="200"></svg>
     </div>
   </div>
-</section>
-<section class="p-strip--light is-bordered">
-  <div class="row u-equal-height">
+  <div class="row is-bordered">
```
Comment (anthonydillon) [inline templates/desktop/statistics.html]: I think using `<hr />` is better here then `is-bordered`

**Comment by anasereijo** [comment]
Hi @inuwa this is starting to look great 👍

I will give design -1 for now because the graphs aren't looking like the ones I designed yet and you also need to update the hero section and the sticky nav too. but it's getting there! :)

thank you!

**Comment by inuwa** [comment]
Comments raised within the PR have been addressed. I noticed from @anasereijo was checking the charts. Apologies if the description is not very clear. It just the layout, that is being addressed in this PR. i.e.

- The number of Sections
- The color shade of the sections (gray or white)
- The number of columns per section
- The number of charts/placeholders per section
- The width of each column in each section
Thank you.

**Comment by spencerbygraves** [comment]
@inuwa can you add a demo link please? Thanks.

**Comment by anthonydillon** [comment]
@spencerbygraves the link at the top of the PR should be up to date.

File: static/js/navigation.js [hunk 11]
```diff
@@ -163,7 +163,7 @@ function closeMainMenu() {
   }
 }
 
-var origin = window.location.href;
+var origin = window.location.pathname;
```
Comment (anthonydillon) [inline static/js/navigation.js]: You need to revert this file change. To do so:
``` bash
git fetch upstream
git checkout upstream/master -- static/js/navigation.js
```

File: templates/base_index.html [hunk 12]
```diff
@@ -112,24 +111,44 @@ <h3><a href="/desktop">Fast, secure and simple, Ubuntu powers millions of PCs wo
...
+            lang = lang || window.navigator.language ||
+            window.navigator.browserLanguage || window.navigator.userLanguage;
 
+            if (lang && lang.length) {
+                return lang;
+            } else {
+                return null
+            }
+        };
+    </script>
```
Comment (anthonydillon) [inline templates/base_index.html]: I assume this was added by rebase'ing on master. This should be done in the future in a separate PR but ok this time.

File: templates/desktop/statistics.html [hunk 13]
```diff
@@ -33,109 +33,138 @@ <h1>Ubuntu user statistics</h1>
...
-      <h2 class="p-heading--four">How many users opted in?</h2>
-      <p>We appreciate the trust that has been given to us with handling this data and we aim to use the results to focus our development efforts on where it matters most to users.</p>
+  <div class="row">
+      <div class="col-12">
+        <h2>User Report</h2>
+      </div>
+  </div>
+  <div class="row u-equal-height u-vertically-center p-divider">
+    <div class="col-6 p-divider__block">
+      <div class="col-12">
```
[the remark this answers is not in the document]
Comment (anthonydillon) [reply templates/desktop/statistics.html]: Please remove

File: templates/desktop/statistics.html [hunk 14]
```diff
@@ -32,214 +29,198 @@ <h1>Ubuntu user statistics</h1>
...
   <div class="row u-equal-height">
-    <svg id="popular-screen-sizes" height="320"></svg>
-  </div>
-</section>
-<section class="p-strip--light is-bordered u-no-padding--top">
-  <div class="row u-equal-height">
+    <div class="col-5">
+      <svg id="popular-screen-sizes" height="320"></svg>
+    </div>
+    <div class="col-1 suffix-1"></div>
```
Comment (anthonydillon) [inline templates/desktop/statistics.html]: suffix adds extra padding inside the current column therefore this should be added to the column above and change that from `col-5` to `col-6`

File: templates/desktop/statistics.html [hunk 15]
```diff
@@ -32,214 +29,198 @@ <h1>Ubuntu user statistics</h1>
...
-    </div>
-    <div class="col-6">
-      <p>Physical disks denote how much data can be stored on a computer. For this report the total disk size is inferred from the number and size of partitions of a disk.</p>
+      <svg id="physical-disk" height="200"></svg>
     </div>
   </div>
-</section>
-<section class="p-strip is-bordered u-no-padding--top">
   <div class="row">
+    <hr />
```
Comment (anthonydillon) [inline templates/desktop/statistics.html]: This row should have a col. Please wrap in `<div class="col-8">`

File: templates/desktop/statistics.html [hunk 16]
```diff
@@ -303,26 +290,16 @@ <h3 class="p-heading--four">Users who chose to download and update software whil
...
-  <div class="row">
-    <div id="how-long-to-get-running">
-      <img src="https://via.placeholder.com/1040x400?text=Area chart" alt="">
-    </div>
-  </div>
-</section>
 <section class="p-strip is-bordered">
-  <div class="row">
+    <div class="row">
+        <div class="col-5 prefix-1">
```
Comment (anthonydillon) [inline templates/desktop/statistics.html]: This should be `col-6`

**Pull request merged**